### ISSUE TITLE
Slow in receiving large numbers of messages

### ISSUE THREAD

**Issue by Zombiekat** [signalapp/Signal-Android#9183]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

Slow in receiving batch messages/notifications. Maybe takes large amount of time to receive all messages especially bulk messages from large groups when we’re offline for some time.

### Steps to reproduce
Send large number of messages and check how notifications are receiving. How much time it takes. 

**Actual result:** 
Slow in receiving large numbers of messages or batch messages.

**Expected result:**
Better handling of notifications when receiving large numbers of messages. 

### Screenshots
https://imgur.com/a/ns4ntuo

### Device info
Redmi 5A
Android 8.1

### Link to debug log
Not a bug. My friend Samsung j7 prime behaviour also same. iOS version has no problems, only in Android.

**Comment by Tursko** [comment]
Link the debug log from the J7.

**Comment by Zombiekat** [comment]
Thanks for responding. I’ve noticed this behaviour on my friend j7. For testing Purpose, I’ve sent myself few messages(30 in chat / 30 in group) with same order in signal as well as threema from my iPhone to my other device redmi 5A. Threema easily managed to receive all notifications within seconds, where signal keeps on receiving one by one. 

My friend is out of reach at this moment. This screen recording is from my device (Redmi 5A). Debug logs are captured seconds after this recorded video. 

I’ve got another android phone Sony Xperia tipo without google play services but the behaviour is same on that device as well. This notification behaviour is same on all 3 android devices I’ve tested.

https://drive.google.com/file/d/1-0Ysoh_gA103RupRatvwOHIZPa5wjiku/view?usp=drivesdk

https://debuglogs.org/55f275cb70c71533dce58b03cbcf1fc62d2a89f658f7d987c21224e05aaea3bc

**Comment by idirouhab** [comment]
I'm actually facing the same problem with my Honor 10.

I usually use the desktop app and then from time to time I shift to the mobile app.
Lately, I had a very long chat using the desktop version and then when I switched to the Android one took 9 minutes until all messages were synced.

The biggest issue here is that while I'm waiting for the messages to get fully synced I used my laptop which is adding even more message making it slower.
